tools/repl/repl.cpp

```diff
@@ -222,6 +228,38 @@ static bool needsAnotherLine(llvm::StringRef input) {
   return !stack.empty();
 }
 
+#if HAVE_LIBREADLINE
+// Load history file or create it
+std::error_code loadOrCreateHistoryFile(llvm::SmallString<128>& historyFile) {
+  llvm::Twine baseHistoryFile = llvm::Twine(HISTORY_FILE);
```

Comment:
`Twine` [should never be stored as a local variable](http://llvm.org/docs/ProgrammersManual.html#the-twine-class), because it may keep a reference to external memory.

`Twine` is cheap to create and should only be constructed as the argument to a function call. It has implicit constructors from things like `const char *` and string literals so you can pass `HISTORY_FILE` directly to `home_directory` and all the other `llvm::sys` functions below.

Reply:
Done